### PR TITLE
Add ability to codegen models and clients for the REST apis.

### PR DESCRIPTION
**Pull request by mikenevi**

### CODE REVIEW

File: MHVLib/Json/MHVJsonSerializer.m [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+#import "MHVJsonEnumProtocol.h"
+#import "MHVJsonCacheRetainObjectProtocol.h"
+#import "MHVDataModelProtocol.h"
+#import "MHVJsonCacheService.h"
+#import "NSString+DataModel.h"
+#import "NSArray+DataModel.h"
+#import "NSDictionary+DataModel.h"
+#import "MHVPersistedDataService.h"
+#import "MHVValidator.h"
+#import "NSObject+MHVDependencyInjection.h"
```
Comment (namalu) [inline MHVLib/Json/MHVJsonSerializer.m]: If you delete #import "NSObject+MHVDependencyInjection.h", and line 155 (MHVASSERT_TRUE([ResolveProtocol(MHVPersistedDataService) fileExistsAtPath:filePath], @"file not found");) you can delete the entire DI folder. We aren't going to use dependency injection in the SDK (as of now). So we should slim down as much as we can.

File: MHVLib/Json/MHVJsonSerializer.m [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+#import "MHVJsonCacheItem.h"
+#import "MHVQueuedDictionary.h"
+#import "MHVJsonEnumProtocol.h"
+#import "MHVJsonCacheRetainObjectProtocol.h"
+#import "MHVDataModelProtocol.h"
+#import "MHVJsonCacheService.h"
+#import "NSString+DataModel.h"
+#import "NSArray+DataModel.h"
+#import "NSDictionary+DataModel.h"
+#import "MHVPersistedDataService.h"
```
Comment (namalu) [inline MHVLib/Json/MHVJsonSerializer.m]: If you delete #import "MHVPersistedDataService.h" you can delete MHVPersistedDataService. We aren't going to use the persisted data service in this project.

File: MHVLib/MHVPersistedDataService.h [hunk 3]
```diff
@@ -0,0 +1,119 @@
+//
+// MHVPersistedDataService.h
```
Comment (namalu) [inline MHVLib/MHVPersistedDataService.h:2]: Per my previous comments, This class can be deleted.

File: MHVLib/Models/Categories/NSString+Utils.m [hunk 4]
```diff
@@ -0,0 +1,415 @@
+//
+// NSString+Utils.m
```
Comment (namalu) [inline MHVLib/Models/Categories/NSString+Utils.m]: Most of these category methods are not used and are very specific to the HVI app. It looks like the only methods being used are '**boolString**' and '**capitalizedStringForSelectors**'. Let's move these 2 methods into MHVStringExtensions, so string extensions are all located in one spot.

File: MHVLib/Models/DynamicConfigurationEnums/MHVCaloriesSystemOfMeasure.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
+//
+// MHVCaloriesSystemOfMeasure.h
```
Comment (namalu) [inline MHVLib/Models/DynamicConfigurationEnums/MHVCaloriesSystemOfMeasure.h]: I believe all of the DynamicConfigurationEnums can be deleted.

File: MHVLib/Models/Generated/MHVActionPlanTasksResponseActionPlanTaskInstance_.h [hunk 6]
```diff
@@ -0,0 +1,49 @@
+//
+// MHVActionPlanTasksResponseActionPlanTaskInstance_.h
```
Comment (namalu) [inline MHVLib/Models/Generated/MHVActionPlanTasksResponseActionPlanTaskInstance_.h:2]: What's the deal with the underscore?

**Review by namalu — APPROVED** [review]

File: MHVLib/Json/MHVJsonSerializer.m (hunk 2 above)
Comment (mikenevi) [reply MHVLib/Json/MHVJsonSerializer.m]: Done

File: MHVLib/Json/MHVJsonSerializer.m (hunk 1 above)
Comment (mikenevi) [reply MHVLib/Json/MHVJsonSerializer.m]: Done

File: MHVLib/Models/DynamicConfigurationEnums/MHVCaloriesSystemOfMeasure.h (hunk 5 above)
Comment (mikenevi) [reply MHVLib/Models/DynamicConfigurationEnums/MHVCaloriesSystemOfMeasure.h]: Yep, deleted.

File: MHVLib/Models/Generated/MHVActionPlanTasksResponseActionPlanTaskInstance_.h (hunk 6 above)
Comment (mikenevi) [reply MHVLib/Models/Generated/MHVActionPlanTasksResponseActionPlanTaskInstance_.h:2]: Not quite sure what this one is about. I think it has to do with the V2 API's being around in the swagger. I'll clean it up next sprint when I get around to testing some of these in more depth.

File: MHVLib/MHVPersistedDataService.h (hunk 3 above)
Comment (mikenevi) [reply MHVLib/MHVPersistedDataService.h:2]: Deleted!

**Pull request merged**